src/passes/OptimizeInstructions.cpp

```diff
@@ -484,6 +484,26 @@ struct OptimizeInstructions
         if (auto* ret = optimizeAddedConstants(binary)) {
           return ret;
         }
+      } else if (binary->op == MulFloat32 || binary->op == MulFloat64 ||
+                 binary->op == DivFloat32 || binary->op == DivFloat64) {
+        if (binary->left->type == binary->right->type) {
+          if (auto* leftUnary = binary->left->dynCast<Unary>()) {
+            if (leftUnary->op ==
+                Abstract::getUnary(binary->type, Abstract::Abs)) {
+              if (auto* rightUnary = binary->right->dynCast<Unary>()) {
+                if (leftUnary->op == rightUnary->op) { // both are abs ops
+                  // abs(x) * abs(y)   ==>   abs(x * y)
+                  // abs(x) / abs(y)   ==>   abs(x / y)
+                  binary->left = leftUnary->value;
+                  binary->right = rightUnary->value;
+                  return Builder(*getModule())
+                    .makeUnary(Abstract::getUnary(binary->type, Abstract::Abs),
+                               binary);
```

Comment:
I think you could reuse `leftUnary` here rather than creating a new node. I also think this would be more readable with matchers even though we can still just match one operation at a time, but I'll leave that up to you.

```
using Abstract; // not sure if this is possible?
if (matches(bin, binary(Mul, unary(&left, Abs), unary(&right, Abs))) ||
    matches(bin, binary(Div, unary(&left, Abs), unary(&right, Abs)))) {
    ...
}
```

Reply:
> I think you could reuse leftUnary here rather than creating a new node

Nice catch! This great optimization.

> using Abstract; // not sure if this is possible?

unfortunately this not possible. Also `binary` expression conflict with `Match::binary`. So I prefer leave this as is